### PR TITLE
pythonPackages.pyqt5: fix sip dependency

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

Related to the discution in #49400

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/pyqt/5.x.nix [hunk 1]
```diff
@@ -65,6 +65,8 @@ in buildPythonPackage {
   '';
 
   postInstall = ''
+    ln -s ${sip}/${python.sitePackages}/sip.pyi $out/${python.sitePackages}/PyQt5
+    ln -s ${sip}/${python.sitePackages}/sip.so $out/${python.sitePackages}/PyQt5
```
Comment (FRidh) [inline pkgs/development/python-modules/pyqt/5.x.nix]: replace `.so` with `${stdenv.hostPlatform.extensions.sharedLibrary}`

**Review by FRidh — DISMISSED** [review]

File: pkgs/development/python-modules/pyqt/5.x.nix (hunk 1 above)
Comment (nyanloutre) [reply pkgs/development/python-modules/pyqt/5.x.nix]: done :+1:

**Comment by timokau** [comment]
@GrahamcOfBorg eval

**Comment by timokau** [comment]
@GrahamcOfBorg eval

**Comment by timokau** [comment]
@GrahamcOfBorg build python2.pkgs.pyqt5 python3.pkgs.pyqt5

**Comment by nyanloutre** [comment]
Sorry I had to rewrite is a bit. I first tried to skip the module rename by only linking to the library but this wasn't enough.

**Comment by timokau** [comment]
@GrahamcOfBorg build python2.pkgs.pyqt5 python3.pkgs.pyqt5

**Comment by timokau** [comment]
@GrahamcOfBorg build calibre

**Comment by veprbl** [comment]
I wonder if `--sip-module` needs to be applied conditionally to preserve compatibility for those who use `sip` without `pyqt5`. Also, perhaps, we should now move `sip` out of `pyqt5.propagatedBuildInputs` to just `pyqt5.buildInputs` so that `sip` can't clash with `pyqt5`. Also `sip` and `pyqt5` currently often go together in dependencies of other packages, we would probably need to eliminate those references to `sip` as well.

**Comment by orivej** [comment]
It seems that `pyqt5` depends on https://pypi.org/project/PyQt5-sip/ rather than on https://pypi.org/project/SIP/.

**Comment by orivej** [comment]
Possible fix: #52897

**Pull request merged**